### ISSUE TITLE
url base path

### ISSUE THREAD

**Issue by dedins** [seripap/darkwire.io#38]
Hi, is it possible to edit the urlbase path for reverse proxy using a path like /darkwire? Thanks

**Comment by seripap** [comment]
Haven't tested this- but in theory, if you're using nginx you can proxy pass an endpoint to the node app.

```
server {
        listen 80;
        location /darkwire {
             proxy_pass http://localhost:3000
        }
}
```

**Comment by dedins** [comment]
I tried setting nginx like that but doesn't work, I think the app has to know that every request must include /darkwire.
I tried also to edit app.js routes to that path but something broke (think static stuff doesn't load)

**Comment by alanfriedman** [comment]
How about [this solution](http://serverfault.com/a/562850)?

Same as above, except add a trailing slash:

    server {
        listen 80;

        location = /darkwire {
            return 302 /darkwire/;
        }

        location /darkwire/ {
             proxy_pass http://localhost:3000
        }
    }

**Comment by vinkenvvt** [comment]
I met this problem too. 

I set up nginx as a reverse proxy https://localhost/darkwire, proxying http://localhost:3000 which places darkwire. 
I've modified `src/js/app.js`, `line: 14` from `this._socket = io(this._roomId);` to `this._socket = io(this._roomId,{path:'/darkwire/socket.io'});`, then websocket is working properly, but darkwire is still not usable.

I've made sure that nginx is going well. Don't know what else I should modify. 

Thanks for any tips.

**Comment by vinkenvvt** [comment]
I met this problem too, but already solved.
Here is what I modified: 
```
src/app.js
  line 39: return res.redirect(`/darkwire/${id}`);
  line 62: return res.redirect('/darkwire/');
src/js/app.js
  line 12: this._roomId = window.location.pathname.length ? '/'+window.location.pathname.split('/').filter(function(el){ return !!el; }).pop() : null;
  line 14: this._socket = io(this._roomId,{path:'/darkwire/socket.io'});
  line 23: $('input.share-text').val(document.location.protocol + '//' + document.location.host + '/darkwire' + this._roomId);
```

**Comment by seripap** [comment]
Will be fixed in #26
